File: 백준/Silver/1629. 곱셈/곱셈.py
a,b,c = map(int, input().split())

def go(a,b):
    if b == 1:
        return a % c
    else:
        x = go(a, b // 2)
        if b % 2 ==0:
            return (x*x) % c
        else:
            return (x*x *a) % c

print(go(a,b))

File: 백준/Silver/1629. 곱셈/곱셈.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;

public class Main {
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    String[] prob = br.readLine().split(" ");

    BigInteger a = new BigInteger(prob[0]);
    BigInteger b = new BigInteger(prob[1]);
    BigInteger c = new BigInteger(prob[2]);

    BigInteger result = a.modPow(b, c);

    System.out.println(result);
  }
}